pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -450,3 +451,38 @@ func (a Adapter) Delete(labels map[string]string) error {
 
 	return a.Client.DeleteDeployment(labels)
 }
+
+// Log returns log from component
+func (a Adapter) Log(follow, debug bool) (io.ReadCloser, error) {
+
+	if !utils.ComponentExists(a.Client, a.ComponentName) {
+		return nil, errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	}
+
+	var command versionsCommon.DevfileCommand
+	var err error
+
+	if debug {
+		command, err = common.GetDebugCommand(a.Devfile.Data, "")
+		if err != nil {
+			return nil, err
+		}
+		if reflect.DeepEqual(versionsCommon.DevfileCommand{}, command) {
+			return nil, errors.Errorf("no debug command found in devfile, please run \"odo log\" for run command logs")
+		}
+
+	} else {
+		command, err = common.GetRunCommand(a.Devfile.Data, "")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	pod, err := a.Client.GetPodUsingComponentName(a.ComponentName)
```

Comment:
we should check the status of the pod too and check if it's running or not.
https://github.com/openshift/odo/blob/e367c334c2e6a15f584821c34af4659bb2658a30/pkg/debug/portforward.go#L71

Reply:
Done.